src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -32,6 +32,31 @@
 #define UCT_RC_MLX5_CHECK_PUT_SHORT(_length, _av_size) \
     UCT_CHECK_LENGTH(_length, 0, UCT_IB_MLX5_PUT_MAX_SHORT(_av_size), "put_short")
 
+#define UCT_RC_MLX5_ATOMIC_OPS (UCS_BIT(UCT_ATOMIC_OP_ADD) | \
+                                UCS_BIT(UCT_ATOMIC_OP_AND) | \
+                                UCS_BIT(UCT_ATOMIC_OP_OR)  | \
+                                UCS_BIT(UCT_ATOMIC_OP_XOR))
+
+#define UCT_RC_MLX5_ATOMIC_FOPS (UCT_RC_MLX5_ATOMIC_OPS | UCS_BIT(UCT_ATOMIC_OP_SWAP))
+
+#define UCT_RC_MLX5_CHECK_ATOMIC_OPS(_op, _size)                                \
+    if (ucs_unlikely(!(UCS_BIT(_op) & UCT_RC_MLX5_ATOMIC_OPS))) {               \
+        ucs_assertv(0, "incorrect opcode for atomic_post: %d", _op);            \
+        return UCS_ERR_UNSUPPORTED;                                             \
+    } else {                                                                    \
+        ucs_assert((_size == sizeof(uint64_t)) || (_size == sizeof(uint32_t))); \
+    }
+
+#define UCT_RC_MLX5_CHECK_ATOMIC_FOPS(_op, _size)                               \
```

Comment:
but the code still almost the same as UCT_RC_MLX5_CHECK_ATOMIC_OPS )

Reply:
ok, updated